### PR TITLE
(FACT-1414) Allow facts/windows.rb to run in all Windows platforms

### PR DESCRIPTION
**Pull request by glennsarti**
Previously the facts/windows.rb acceptance tests would only run on
Windows 2003, 2008 and 2012.  This commit;
- Relaxes the confinement to any Windows platform
- Adds OS and Kernel version checks for Windows 10
- Fixed a typo in the kernel_version checks. These are regular expressions
so the '.' was not literal e.g. `6.3` should be `6\.3`
- Tests will fail if an unknown Windows platform is encountered instead of being
skipped.  This will ensure that new platforms must be configured instead of
silently skipping.

Note - This PR superceeds https://github.com/puppetlabs/facter/pull/1274

### CODE REVIEW

File: acceptance/tests/facts/windows.rb [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
 #
 # This test is intended to ensure that facts specific to an OS configuration
-# resolve as expected in Windows 2003R2, 2008R2 and 2012R2.
+# resolve as expected on Windows platforms.
 #
 # Facts tested: os, processors, networking, identity, kernel
 #
 
-confine :to, :platform => /windows-2003r2|windows-2008|windows-2012r2/
+confine :to, :platform => /windows-/
```
Comment (ferventcoder) [inline acceptance/tests/facts/windows.rb]: 👍

**Comment by ferventcoder** [comment]
Nothing looks out of the ordinary here.

**Comment by ferventcoder** [comment]
Commits look good as well.

**Comment by ferventcoder** [comment]
@puppetlabs/windows reviews are good. 👍

**Comment by Iristyle** [comment]
This all looks 👍 to me... I assume it passed acceptance when run locally?

**Comment by glennsarti** [comment]
Local acceptance tests passed for Win10 x86/x64 and 2012R2 x64

File: acceptance/.gitignore [hunk 2]
```diff
@@ -9,3 +9,4 @@ preserved_config.yaml
 merged_options.rb
 junit
 tmp
+.bundle
```
Comment (Iristyle) [inline acceptance/.gitignore]: So close...!

**Comment by Iristyle** [comment]
Ready to merge once the `.gitignore` newline is straightened out.

File: acceptance/.gitignore (hunk 2 above)
Comment (glennsarti) [inline acceptance/.gitignore]: Fixed

**Pull request merged**